src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Conditions;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        { 
+            IEnumerable<IElement> elements = page.ValidatableElements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIf?.QuestionId));
+
+            if (!elements.Any())
+                return await Task.FromResult(page);
+
+            foreach (IElement element in elements)
+            {
+                var userChoice = convertedAnswers.AllAnswers
+                    .FirstOrDefault(answer => answer.QuestionId
+                    .Equals(element.Properties.OptionalIf.QuestionId));
+
+                var questionsAnswers = new Dictionary<string, dynamic>
+                {
+                    { userChoice.QuestionId, userChoice.Response }
+                };
+
+                var conditionValidator = new ConditionValidator();
+                element.Properties.Optional = conditionValidator.IsValid(element.Properties.OptionalIf, questionsAnswers);
```

Comment:
Could be simplified further... could be just
  element.Properties.Optional = new ConditionValidator()
                    .IsValid(element.Properties.OptionalIf, new() { { answer.QuestionId, answer.Response } });
Which hopefully saves the need to think about making names up for two more variables.

Reply:
Just to counter Stu's suggestion, I'd be inclined to leave the Dictionary variable as is (better readability to me at least), but change lines 31/32 to Stu's suggestion using the Dictionary variable rather than the new()